src/App.tsx

```diff
@@ -1,26 +1,198 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { deleteTodo, getTodos, postTodo, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { sendErrorMessage } from './utils/sendError';
+import { useRef } from 'react';
+import { Footer } from './components/footer';
+import { ErrorMessage } from './components/errorsUnderFooter';
+import { MainSection } from './components/section';
+import { Header } from './components/header';
 
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [filteredTodoList, setFilteredTodoList] = useState<Todo[]>([]);
+  const [selectedFilter, setSelectedFilter] = useState('all');
+  const [title, setTitle] = useState('');
+  const [loader, setLoader] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const [deletingTodos, setDeletingTodos] = useState<number[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(data => {
+        setTodoList(data);
+        setFilteredTodoList(data);
+      })
+      .catch(() => {
+        sendErrorMessage('Unable to load todos', setErrorMessage);
```

Comment:
It's enouth here to use errorMessage state as you can pass it to error component as well